geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/sortedset/AbstractZPopMinIntegrationTest.java

```diff
@@ -0,0 +1,157 @@
+  }
+
+  @After
+  public void tearDown() {
+    flushAll();
+    jedis.close();
+  }
+
+  @Test
+  public void shouldError_givenWrongNumberOfArguments() {
+    assertThatThrownBy(
+        () -> jedis.sendCommand("key", Protocol.Command.ZPOPMIN, "key", "1", "2"))
+            .hasMessageContaining(RedisConstants.ERROR_SYNTAX);
+  }
+
+  @Test
+  public void shouldError_givenWrongNumberFormat() {
+    assertThatThrownBy(
+        () -> jedis.sendCommand("key", Protocol.Command.ZPOPMIN, "key", "wat"))
+            .hasMessageContaining(RedisConstants.ERROR_NOT_INTEGER);
+  }
+
+  @Test
+  public void shouldReturnEmpty_givenNonExistentSortedSet() {
+    assertThat(jedis.zpopmin("unknown", 1)).isEmpty();
+  }
+
+  @Test
+  public void shouldReturnEmpty_givenNegativeCount() {
+    jedis.zadd("key", 1, "player1");
+
+    List<?> result = (List<?>) jedis.sendCommand("key", Protocol.Command.ZPOPMIN, "key", "-1");
+    assertThat(result).isEmpty();
+    assertThat(jedis.zrange("key", 0, 10)).containsExactly("player1");
+  }
+
+  @Test
+  public void shouldReturn_highestLex_whenScoresAreEqual() {
+    jedis.zadd("key", 1, "player1");
+    jedis.zadd("key", 1, "player2");
+    jedis.zadd("key", 1, "player3");
+
+    assertThat(jedis.zpopmin("key").getElement()).isEqualTo("player1");
+    assertThat(jedis.zrange("key", 0, 10))
+        .containsExactlyInAnyOrder("player2", "player3");
+  }
+
+  @Test
+  public void shouldReturn_lowestScore() {
+    jedis.zadd("key", 3, "player1");
+    jedis.zadd("key", 2, "player2");
+    jedis.zadd("key", 1, "player3");
+
+    assertThat(jedis.zpopmin("key").getElement()).isEqualTo("player3");
+    assertThat(jedis.zrange("key", 0, 10))
+        .containsExactlyInAnyOrder("player1", "player2");
+  }
+
+  @Test
+  public void withCountShouldReturn_membersInScoreOrder_whenScoresAreDifferent() {
+    List<Tuple> tuples = new ArrayList<>();
+    int count = 10;
+    // Make sure that the results are not somehow dependent on the order of insertion
+    List<Integer> shuffles = makeShuffledList(count);
+    for (int i = 0; i < count; i++) {
+      jedis.zadd("key", count - shuffles.get(i), "player" + shuffles.get(i));
+      tuples.add(new Tuple("player" + (count - i - 1), (double) (i + 1)));
+    }
+
+    assertThat(jedis.zpopmin("key", count)).containsExactlyElementsOf(tuples);
+  }
+
+  @Test
+  public void withCountShouldReturn_membersInReverseLexicalOrder_whenScoresAreTheSame() {
+    List<Tuple> tuples = new ArrayList<>();
+    int count = 10;
+    // Make sure that the results are not somehow dependent on the order of insertion
+    List<Integer> shuffles = makeShuffledList(count);
+    for (int i = 0; i < count; i++) {
+      jedis.zadd("key", 1D, "player" + shuffles.get(i));
+      tuples.add(new Tuple("player" + i, 1D));
+    }
+
+    assertThat(jedis.zpopmin("key", count)).containsExactlyElementsOf(tuples);
+  }
+
+  @Test
+  public void shouldReturn_countLowestScores() {
```

Comment:
this test name also doesn't quite make sense to me

Reply:
Hopefully more descriptive now.